lucene/core/src/test/org/apache/lucene/document/TestManyKnnDocs.java

```diff
@@ -46,27 +54,139 @@ public void testLargeSegment() throws Exception {
     mp.setMaxMergeAtOnce(256); // avoid intermediate merges (waste of time with HNSW?)
     mp.setSegmentsPerTier(256); // only merge once at the end when we ask
     iwc.setMergePolicy(mp);
-    String fieldName = "field";
-    VectorSimilarityFunction similarityFunction = VectorSimilarityFunction.DOT_PRODUCT;
+    VectorSimilarityFunction similarityFunction = VectorSimilarityFunction.EUCLIDEAN;
```

Comment:
Could we retain use of existing similarity function? And if alternative is needed perhaps have that for an additional field?
```suggestion
    VectorSimilarityFunction similarityFunction = VectorSimilarityFunction.DOT_PRODUCT;
```